pkg/artifacts/signable.go

```diff
@@ -175,6 +154,78 @@ func ExtractOCIImagesFromResults(tr *v1beta1.TaskRun, logger *zap.SugaredLogger)
 	return objs
 }
 
+// ExtractIntotoSignableTargetFromResults extracts signable targets that aim to generate intoto provenance as materials within TaskRun results and store them as structuredSignable.
+func ExtractIntotoSignableTargetFromResults(tr *v1beta1.TaskRun, logger *zap.SugaredLogger) []interface{} {
+	return extractTargetFromResults(tr, "INTOTO_TARGET_NAME", "INTOTO_TARGET_DIGEST", logger, false)
+}
+
+func extractTargetFromResults(tr *v1beta1.TaskRun, identifierSuffix string, digestSuffix string, logger *zap.SugaredLogger, isOCIImage bool) []interface{} {
+	ss := map[string]*StructuredSignable{}
+	var objs []interface{}
+
+	for _, res := range tr.Status.TaskRunResults {
+		if strings.HasSuffix(res.Name, identifierSuffix) {
+			marker := strings.TrimSuffix(res.Name, identifierSuffix)
+			if v, ok := ss[marker]; ok {
+				v.Name = strings.TrimSpace(res.Value.StringVal)
+
+			} else {
+				ss[marker] = &StructuredSignable{Name: strings.TrimSpace(res.Value.StringVal)}
+			}
+			// TODO: add logic for Intoto signable target as input.
+		}
+		if strings.HasSuffix(res.Name, digestSuffix) {
+			marker := strings.TrimSuffix(res.Name, digestSuffix)
+			if v, ok := ss[marker]; ok {
+				v.Digest = strings.TrimSpace(res.Value.StringVal)
+			} else {
+				ss[marker] = &StructuredSignable{Digest: strings.TrimSpace(res.Value.StringVal)}
+			}
+		}
+
+	}
+
+	// Only add it if we got both the artifact name and digest.
+	// TODO: add logic for adding signable target as input.
+	for _, s := range ss {
+		if s != nil && s.Digest != "" {
```

Comment:
I'm finding the multiple nested if-else-blocks difficult to read. Consider reversing some of the checks and using `continue` to reduce nesting. Maybe something like this (untested):
```go
if s == nil || s.Digest == "" {
  continue
}
if err := checkDigest(s.Digest); err != nil {
  logger.Errorf("error getting digest %s: %v", s.Digest, err)
  continue
}

var newObj interface{}
newObj = s
if s.Name != "" && isOCIImage {
  dgst, err := name.NewDigest(fmt.Sprintf("%s@%s", s.Name, s.Digest))
  if err != nil {
    logger.Errorf("error getting digest: %v", err)
    continue
  }
  newObj = dgst
}
objs = append(objs, newObj)
```

Reply:
Sure!